lib/engine/step/tracker.rb

```diff
@@ -71,14 +71,22 @@ def lay_tile(action, extra_cost: 0, entity: nil)
         discount = 0
 
         entity.abilities(:tile_lay) do |ability|
-          next if !ability.hexes.include?(hex.id) || !ability.tiles.include?(tile.name)
+          next unless ability.hexes.empty?
+
+          corporation = entity.owner
+          @game.game_error("Track laid in must be connected to one of #{corporation.id}'s stations") if
+            !@game.graph.reachable_hexes(corporation)[hex] && hex.name != corporation.coordinates
+        end if entity.company?
+
+        entity.abilities(:tile_lay) do |ability|
+          next if ability.hexes.any? && (!ability.hexes.include?(hex.id) || !ability.tiles.include?(tile.name))
```

Comment:
same here

Reply:
I think it is correct, after the change of the code. We do want to use free and discount ability attributes if hexes is empty. 
In case of the tile_lay in 1882 it will mean that the following code is exercised, which is wasn't before but the NWR does not specify free or discount, so it will use default which is free=false and discount=0, so it wont change anything.

So this should be OK.